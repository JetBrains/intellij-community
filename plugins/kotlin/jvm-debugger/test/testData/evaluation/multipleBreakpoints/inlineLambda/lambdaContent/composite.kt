// IGNORE_K1

fun main() {
    foo { it ->

        fun localFunWithLocalClass(p: Int): Int {
            class X(val x: Int)
            return X(p).x
        }

        fun createFunWithLocalObj(): () -> Int {
            return {
                object {
                    val x = 1
                }.x
            }
        }

        val complexLambda = object {
            val prop = {

                var x = 10
                x += 100

                val y = localFunWithLocalClass(1000)

                val z = createFunWithLocalObj()()

                val l = { 10000 }

                x + y + z + l()
            }
        }.prop

        complexLambda() + it
    }
}

inline fun foo(block: (Int) -> Int) {
    // EXPRESSION: block(100000)
    // RESULT: 111111: I
    //Breakpoint!
    val x = 1
}