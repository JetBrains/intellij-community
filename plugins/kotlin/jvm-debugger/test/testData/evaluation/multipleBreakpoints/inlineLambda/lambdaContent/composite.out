LineBreakpoint created at composite.kt:43
Run Java
Connected to the target VM
composite.kt:43
Compile bytecode for block(100000)
Disconnected from the target VM

Process finished with exit code 0
